components/contribution-flow/PlatformTipContainer.tsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+
+import { formatCurrency } from '../../lib/currency-utils';
+import { Currency } from '../../lib/graphql/types/v2/graphql';
+import theme from '../../lib/theme';
+
+import { Box, Flex } from '../Grid';
+import { I18nBold } from '../I18nFormatters';
+import Image from '../Image';
+import StyledButtonSet from '../StyledButtonSet';
+import StyledHr from '../StyledHr';
+import StyledInputAmount from '../StyledInputAmount';
+import StyledLinkButton from '../StyledLinkButton';
+import { P } from '../Text';
+
+import { WhyPlatformTipModal } from './WhyPlatformTipModal';
+
+const I18nMessages = defineMessages({
+  NO_THANKS: {
+    id: 'NoThankYou',
+    defaultMessage: 'No thank you',
+  },
+  OTHER: {
+    id: 'platformFee.Other',
+    defaultMessage: 'Other',
+  },
+});
+
+export const enum PlatformTipOption {
+  NONE = 'NONE',
+  TEN_PERCENT = 'TEN_PERCENT',
+  FIFTEEN_PERCENT = 'FIFTEEN_PERCENT',
+  TWENTY_PERCENT = 'TWENTY_PERCENT',
+  OTHER = 'OTHER',
+}
+
+type PlatformTipInputProps = {
+  value: number;
+  selectedOption: PlatformTipOption;
+  amount: number;
+  currency: string;
+  onChange: (selectedOption: PlatformTipOption, value?: number) => void;
+  disabled: boolean;
+};
+
+function PlatformTipInput(props: PlatformTipInputProps) {
+  const intl = useIntl();
+
+  const options = React.useMemo(
+    () => ({
+      [PlatformTipOption.NONE]: {
+        label: intl.formatMessage(I18nMessages.NO_THANKS),
+        percent: 0,
+      },
+      [PlatformTipOption.TEN_PERCENT]: {
+        label: '10%',
+        percent: 0.1,
+      },
+      [PlatformTipOption.FIFTEEN_PERCENT]: {
+        label: '15%',
+        percent: 0.15,
+      },
+      [PlatformTipOption.TWENTY_PERCENT]: {
+        label: '20%',
+        percent: 0.2,
+      },
+      [PlatformTipOption.OTHER]: {
+        label: intl.formatMessage(I18nMessages.OTHER),
+        percent: 0.15,
+      },
+    }),
+    [intl],
+  );
+
+  const onOptionChange = React.useCallback(
+    (value: PlatformTipOption) => {
+      if (props.selectedOption === value) {
+        return;
+      }
+
+      props.onChange(value, options[value].percent * props.amount);
+    },
+    [props.onChange, props.selectedOption, props.amount, options],
+  );
+
+  const onOtherChange = React.useCallback(
+    value => {
+      props.onChange(props.selectedOption, value);
+    },
+    [props.onChange],
+  );
+
+  React.useEffect(() => {
+    const newTipAmount =
+      props.selectedOption === PlatformTipOption.OTHER
+        ? props.value
+        : options[props.selectedOption].percent * props.amount;
+
+    props.onChange(props.selectedOption, newTipAmount);
+  }, [options, props.amount, props.value, props.selectedOption]);
+
+  return (
+    <Box data-cy="platform-tip-input">
+      <StyledButtonSet
```

Comment:
Like @Betree pointed out, this breaks on mobile. Flex-direction should be set to column on mobile breakpoints.